pyDataverse/api.py

```diff
@@ -216,19 +215,22 @@ def put_request(self, url, data=None, auth=False, params=None):
         if isinstance(data, str):
             data = json.loads(data)
 
+        # Decide whether to use 'data' or 'json' args
+        request_params = self._check_json_data_form(data)
```

Comment:
I was thinking about calling it with `Api._check_json_data_form` since it's declared `@staticmethod`, but `self` works of course as well and allows for better inheritance.
Maybe it makes sense to move it outside of the Api class? Makes it also much easier to (unit-)test it in isolation.

But I guess this is more or less a matter of personal preference and thus I'd keep it as is. This way it's encapsulated within the API and it would be awkward to call it from other contexts, which I think is good.